crypto/fipsmodule/rsa/rsa.c

```diff
@@ -836,6 +836,15 @@ void RSA_blinding_off_temp_for_accp_compatibility(RSA *rsa) {
   }
 }
 
+int RSA_pkey_ctx_ctrl(EVP_PKEY_CTX *ctx, int optype, int cmd, int p1, void *p2) {
+  if (ctx != NULL && ctx->pmeth != NULL
```

Comment:
* When `ctx` or `ctx->pmeth` are NULL should the function return `0` to match the behavior of `EVP_PKEY_CTX_ctrl`: https://github.com/aws/aws-lc/blob/0541314f0c8ab0e6e4955e70cb31747732797a9c/crypto/fipsmodule/evp/evp_ctx.c#L210 ?
* I find this easier to read with a positive check on key type:
```c++
int RSA_pkey_ctx_ctrl(EVP_PKEY_CTX *ctx, int optype, int cmd, int p1, void *p2) {
  if (ctx != NULL && ctx->pmeth != NULL) {
      if(ctx->pmeth->pkey_id == EVP_PKEY_RSA || ctx->pmeth->pkey_id == EVP_PKEY_RSA_PSS) {
        return EVP_PKEY_CTX_ctrl(ctx, -1, optype, cmd, p1, p2);
      }
      return -1;
  }
  return 0;
}
```

Reply:
> When ctx or ctx->pmeth are NULL should the function return 0 to match the behavior of EVP_PKEY_CTX_ctrl?

I'd considered this, but decided to match OpenSSL's behavior of returning -1 as the goal of adding this function is increasing compatibility there. We shouldn't use this function anywhere internally, and we have moved to a different "semantics" for return values in control functions (OSSL uses negative values for failure, we don't).

> I find this easier to read with a positive check on key type:

Sure, I can adjust this.